earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -329,12 +368,22 @@ ReadDocumentFromString(khxml::DOMLSParser *parser,
 bool
 DestroyDocument(khxml::DOMDocument *doc) throw()
 {
+  bool retval = false;
   try {
     doc->release();
-    return true;
+    retval = true;
   } catch (...) {
+    notify(NFY_WARN, "Unable to release document.");
   }
-  return false;
+
+  if (doTerminate) {
+    XMLPlatformUtils::Terminate();
```

Comment:
I'm not sure I like this pattern as it stands. Functionally, I think it's fine. However, it seems that it would be cleaner if we did the following:

- Don't call initialize in UsingXMLGuard
- Call initialize when creating a DOMDocument
- Call terminate when destroying a DOMDocument

However, as I said, that ends up being functionally the same as this. It's more of a style issue.